### ISSUE TITLE
Edit mode example

### ISSUE THREAD

**Issue by OmidID** [AvaloniaUI/Avalonia.Controls.TreeDataGrid#99]
Hello,

Is there any example that exists for the edit mode?

I try to use TextColumn with getter and setter and no success.

```csharp
new TextColumn<BsonValue, string>(
	key,
	v => v[key].ToString(),
	(b, v) => b[key] = v)
```

Thanks.

**Comment by danwalmsley** [comment]
Not yet supported, readonly for now.

**Comment by jp2masa** [comment]
Double click doesn't work because the current logic is that if a cell is selected, a single click should open edit mode. Unfortunately, I think no logic for cell selection was implemented (there is some for row selection, none for cell).

More details here:

https://github.com/AvaloniaUI/Avalonia.Controls.TreeDataGrid/blob/babd474c9391e9faa55545c223e366d0fd2f130f/src/Avalonia.Controls.TreeDataGrid/Primitives/TreeDataGridCell.cs#L124

And here (this property is never set):

https://github.com/AvaloniaUI/Avalonia.Controls.TreeDataGrid/blob/babd474c9391e9faa55545c223e366d0fd2f130f/src/Avalonia.Controls.TreeDataGrid/Primitives/TreeDataGridCell.cs#L13-L17

https://github.com/AvaloniaUI/Avalonia.Controls.TreeDataGrid/blob/babd474c9391e9faa55545c223e366d0fd2f130f/src/Avalonia.Controls.TreeDataGrid/Primitives/TreeDataGridCell.cs#L32-L36

**Comment by grokys** [comment]
Fixed by #162
